### PR TITLE
Ngfw-14811 IPSEC L2TP logging issue

### PR DESCRIPTION
**Pull request by VarshaKache**
_**Scenerios to test for L2TP service:**_

- When IPSEC is enable and VPN config is disable : L2TP service should not be running.
- When IPSEC is enable and VPN config is enable : L2TP service should be running.
- When IPSEC is disable and VPN Config is enable : L2TP service should not be running.
- When IPSEC made disable : L2TP service should not be running.
- When IPSEC is installed : L2TP should not be running.
- When IPSEC made enable when installed: L2TP should not be running.

### CODE REVIEW

**Comment by VarshaKache** [comment]
Test scenario :
When IPSec is disable :
image
L2TP is not running

File: ipsec-vpn/hier/usr/share/untangle/bin/ipsec-restart-l2tp [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+# Restart the xlt2pd daemon when VPN is Enabled
+# Stop if Disabled
+
+if [ "$1" = "VPN_ENABLED" ]; then
+    systemctl reset-failed xl2tpd 2>&1 1>> /dev/null
+    systemctl restart xl2tpd 2>&1 1>> /dev/null
+
+else
+    systemctl stop xl2tpd 2>&1 1>> /dev/null
```
Comment (singhrohit23) [inline ipsec-vpn/hier/usr/share/untangle/bin/ipsec-restart-l2tp:11]: add this command before 
systemctl reset-failed xl2tpd 2>&1 1>> /dev/null

File: ipsec-vpn/hier/usr/share/untangle/bin/ipsec-restart-l2tp [hunk 2]
```diff
@@ -0,0 +1,15 @@
+#!/bin/dash
+
+# Restart the xlt2pd daemon when VPN is Enabled
+# Stop if Disabled
```
Comment (singhrohit23) [inline ipsec-vpn/hier/usr/share/untangle/bin/ipsec-restart-l2tp:4]: This script is not needed , handle in parent script

File: ipsec-vpn/src/com/untangle/app/ipsec_vpn/IpsecVpnManager.java [hunk 3]
```diff
@@ -151,6 +160,20 @@ public void generateConfig(IpsecVpnSettings settings, String ipsecCertFile)
...
+        boolean isAppEnabled = false;
+
+        if (ipsecVpn != null && ipsecVpn.getAppSettings() != null) {
+            isAppEnabled = ipsecVpn.getAppSettings().getTargetState().equals(AppState.RUNNING);
+        }
+        
+        //xl2tpd should start when the IPsec app is running and the VPN flag is enabled.
+        if(isAppEnabled){
+            UvmContextFactory.context().execManager().exec(RESTART_L2TP_SCRIPT + " " + vpnState) ;
+        }
```
Comment (singhrohit23) [inline ipsec-vpn/src/com/untangle/app/ipsec_vpn/IpsecVpnManager.java:175]: use directly settings.getVpnflag()

**Review by cblaise-untangle — APPROVED** [review]

File: ipsec-vpn/hier/usr/share/untangle/bin/ipsec-restart-l2tp (hunk 1 above)
Comment (VarshaKache) [reply ipsec-vpn/hier/usr/share/untangle/bin/ipsec-restart-l2tp]: Done

**Review by singhrohit23 — APPROVED** [review]

**Pull request merged**